operator/pkg/certs/certs.go

```diff
@@ -428,7 +429,8 @@ func ParsePrivateKeyPEM(keyData []byte) (crypto.Signer, error) {
 	case *ecdsa.PrivateKey:
 		key = k
 	default:
-		return nil, errors.New("the private key is neither in RSA nor ECDSA format")
+		keyType := fmt.Sprintf("%T", caPrivateKey)
```

Comment:
Can you provide an example of what the value of `KeyType` would be when the algorithm is not `RSA` or `ECDSA`?

Reply:
The `vendor/k8s.io/client-go/util/keyutil` package currently supports returning either `RSA` or `ECDSA` keys, or an error if the key type is not recognized. It does not include support for other key types such as `Ed25519`, though new types may be added in the future. I’ve incorporated this as a defensive measure to handle any potential new key types that might be introduced in the upstream package. What do you think?

https://github.com/karmada-io/karmada/blob/ec0521a754b19043c7f5ef946fbdc575e8c2afc5/vendor/k8s.io/client-go/util/keyutil/key.go#L168-L184

Reply:
Also for more info about the algorithms could be used it could be `UnknownPublicKeyAlgorithm`, `RSA`, `DSA`, `ECDSA`, or `Ed25519` as appear in the [Go standard library crypto package](https://pkg.go.dev/crypto/x509#PublicKeyAlgorithm).

Reply:
>  I’ve incorporated this as a defensive measure to handle any potential new key types that might be introduced in the upstream package. What do you think?

LGTM, and as the type of  `caPrivateKey` is `interface{}`, `%T` should be the best way to represent the type of caPrivateKey.